src/python/pants/engine/rules.py

```diff
@@ -90,15 +90,15 @@ def _generate_ast_error_message(self, node, msg):
   class YieldVisitError(Exception): pass
 
   @staticmethod
-  def _maybe_end_of_stmt_list(attr_value):
+  def _maybe_end_of_stmt_list(attr_value: Optional[Any]) -> Optional[Any]:
```

Comment:
Is this `Any` because it should be `Any` or because we don't yet know a more precise type? If the latter, it's better to have this be `Optional`, which MyPy converts to `Optional[Any` but is a signal to us that we can still make things more precise. This is the same idea as `List` vs. `List[Any]`.

Reply:
This is removed in the followup PR to remove rules with `yield`, but your advice is noted for the future!